### ISSUE TITLE
unreachable_pub lint false positive

### ISSUE THREAD

**Issue by lara-rium** [rust-lang/rust#101169]
I tried this code:

```rust
#![warn(unreachable_pub)]

mod foo {
    pub struct Foo;
}

use foo::Foo;
```
[Here's a link for the same code on playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=9f3fee55f1c272893d91b9ab645d79ee)

I expected to see this happen: *No warning, as the item is actually accessible from the root crate*

Instead, this happened: *It says the item is unreachable, even though it isn't*

**Comment by PatchMixolydic** [comment]
`unreachable_pub` trips when a `pub` item would be unreachable from a different crate using the current crate as a dependency, even if it can be reached in the current crate. The solution here is to either mark `Foo` as `pub(crate)` or allow `unreachable_pub` (which is nice for binaries where the distinction between `pub` and `pub(crate)` doesn't matter as much).

**Comment by laralove143** [comment]
This is for a library but nothing in this module should be public, I prefer this way and if I do pub(crate) clippy complains with https://rust-lang.github.io/rust-clippy/master/index.html#redundant_pub_crate so these lints are conflicting? Which one should I listen

**Comment by PatchMixolydic** [comment]
It seems like [that's is a known bug in `clippy::redundant_pub_crate`](https://github.com/rust-lang/rust-clippy/issues/5369), which caused the lint to be moved to the [`clippy::nursery`](https://doc.rust-lang.org/nightly/clippy/index.html) category.
